storage/innobase/trx/trx0i_s.cc

```diff
@@ -645,7 +645,7 @@ put_nth_field(
 
 	/* now buf_size >= 1 */
 
-	data = rec_get_nth_field(rec, offsets, n, &data_len);
+	data = rec_get_nth_cfield(rec, offsets, n, index, NULL, &data_len);
```

Comment:
This change is unnecessary and should be removed.
The reason is that key columns can never be instantly added, and the only caller `fill_lock_data()` is restricting the `n_fields` to the key columns. Thus, `put_nth_field()` can safely assume that all the fields are materialized in the record.